src/app/core/feedback/feedback.controller.js

```diff
@@ -7,90 +7,204 @@ const
+		columns.forEach((col) => {
+			col.callback =
+				col.key === 'created' || col.key === 'updated'
+					? dateCallback
+					: defaultCallback;
 		});
 
-		exportConfigController.exportCSV(req, res, exportFileName, exportColumns, feedbackResult.results);
+		const query = result.config.q ? JSON.parse(result.config.q) : null;
+		const sortArr = [
+			{ property: result.config.sort, direction: result.config.dir }
+		];
+
+		const feedbackResult = await Feedback.textSearch(
+			query,
+			null,
+			null,
+			null,
+			sortArr,
+			true,
+			{
+				path: 'creator',
+				select: ['username', 'organization', 'name', 'email']
+			}
+		);
+
+		exportConfigController.exportCSV(
+			req,
+			res,
+			exportFileName,
+			columns,
+			feedbackResult.results
+		);
 	} catch (err) {
-		logger.error({err: err, req: req}, 'Error exporting feedback entries');
+		logger.error(
+			{ err: err, req: req },
+			'Error exporting feedback entries'
+		);
 		utilService.handleErrorResponse(res, err);
 	}
 };
 
 module.exports.search = async (req, res) => {
-	let query = req.body.q || {};
-	query = { '$and': [ query ] };
+	const filters = [];
+	if (req.body.q) {
+		filters.push(req.body.q);
+	}
 
 	const search = req.body.s || null;
 
 	if (search) {
-		query.$and.push({ $text: { $search: search }});
-		// query.$and.push({ name: new RegExp(search, 'i') });
+		filters.push({ $text: { $search: search } });
 	}
 
+	const query = filters.length > 0 ? { $and: filters } : undefined;
+
+	try {
+		const searchPromise = feedbackService.search(
+			req.user,
+			req.query,
+			query
+		);
+
+		res.status(200).json(await searchPromise);
+	} catch (err) {
+		logger.error(
+			{ err: err, req: req },
+			'Error searching for feedback entries'
+		);
+		utilService.handleErrorResponse(res, err);
+	}
+};
+
+module.exports.updateFeedbackAssignee = async (req, res) => {
+	try {
+		// Audit feedback assignee update
+		await auditService.audit(
+			'Feedback assignee updated',
+			'feedback',
+			'update',
+			TeamMember.auditCopy(
+				req.user,
+				utilService.getHeaderField(req.headers, 'x-real-ip')
+			),
+			req.body,
+			req.headers
+		);
+
+		res.status(200).json(
+			await feedbackService.updateFeedbackAssignee(
+				req.params.feedbackId,
+				req.body.assignee
+			)
+		);
+	} catch (err) {
+		logger.error(
+			{ err: err, req: req },
+			'Error updating feedback assignee'
+		);
+		utilService.handleErrorResponse(res, err);
+	}
+};
+
+module.exports.updateFeedbackStatus = async (req, res) => {
 	try {
-		res.status(200).json(await feedbackService.search(req.user, req.query, query));
+		// Audit feedback status update
+		await auditService.audit(
+			'Feedback status updated',
+			'feedback',
+			'update',
+			TeamMember.auditCopy(
+				req.user,
+				utilService.getHeaderField(req.headers, 'x-real-ip')
+			),
+			req.body,
+			req.headers
+		);
+
+		res.status(200).json(
+			await feedbackService.updateFeedbackStatus(
```

Comment:
@sshah-asymmetrik let's move this service call into a variable that's passed in to the `json(...)` method. That may read easier and let us extend the functionality around that service call moving forward.

Reply:
Move service calls into promises